Source/monster.cpp

```diff
@@ -4411,13 +4413,11 @@ void SyncMonsterAnim(Monster &monster)
 		LevelMonsterTypes[monster.levelType].corpseId = 1;
 	}
 #endif
-	if (monster.uniqType != 0)
-		monster.name = pgettext("monster", UniqueMonstersData[monster.uniqType - 1].mName).data();
-	else
-		monster.name = pgettext("monster", monster.data().mName).data();
-
-	if (monster.uniqType != 0)
+	if (monster.isUnique()) {
+		monster.name = pgettext("monster", UniqueMonstersData[static_cast<size_t>(monster.uniqueType)].mName).data();
 		InitTRNForUniqueMonster(monster);
+	} else
+		monster.name = pgettext("monster", monster.data().mName).data();

```

Comment:
Not a fan of mixed braces
```suggestion
	} else {
		monster.name = pgettext("monster", monster.data().mName).data();
	}
```